src/expressions/capacity_margin.jl

```diff
@@ -0,0 +1,182 @@
+#############################################################################
+
+@doc raw"""
+    add_expression_capacity_margin!(m::Model)
+
+Create an expression for `capacity_margin`. This represents the loat that must be met by conventional
+    resources net of variable renewable production and storage. It is used in the `min_capacity_margin` constraint
+
+```math 
+\begin{aligned}
+expr^{capacity\_margin}_{n,s,t} = \\
+& + \sum_{u\in{U_{n\_to}}}(p^{unit\_capacity}_{u,s,t} \cdot p^{unit\_availability\_factor}_{u,s,t} \cdot v^{units\_available}_{u,s,t}) \\
+& + \sum_{u\in{U_{storage_n}}}(v^{unit\_flow}_{u,n,to,s,t}) \\
+& - \sum_{u\in{U_{storage_n}}}(v^{unit\_flow}_{u,n,from,s,t}) \\
+& - p^{demand}_{n,s,t} \\
+& - p^{fractional\_demand}_{n,s,t} \cdot p^{group\_demand}_{n_{group},s,t} \\
+& \forall n \in node: p^{min\_capacity\_margin} \\
+\end{aligned}
+```
+where ```math U_{storage_n} ``` is the set of all storage units connected to node n \\
+and ```math U_{n\_to} ``` is the set of all non-storage units connected to node n \\
+
+See also
+[min\_capacity\_margin](@ref),
+[min\_capacity\_margin\_penalty](@ref),
+[unit\_\_from\_node](@ref),
+[unit\_\_to\_node](@ref),
+[demand](@ref),
+[fractional\_demand](@ref),
+[has\_state](@ref)
+
+"""
+
+function add_expression_capacity_margin!(m::Model)
+    @fetch unit_flow, units_available = m.ext[:spineopt].variables
+    t0 = _analysis_time(m)
+    
+    m.ext[:spineopt].expressions[:capacity_margin] = Dict(
+        (node=n, stochastic_path=s, t=t) => @expression(
+            m,
+            - sum(                
+                + demand[
+                    (node=n, stochastic_scenario=s, analysis_time=t0, t=first(representative_time_slice(m, t)))
+                ]                
+                for (n, s, t) in node_injection_indices(
+                    m; node=n, stochastic_scenario=s, t=t, temporal_block=anything
+                );
+                init=0,
+            )
+            - sum(
+                fractional_demand[
+                    (node=n, stochastic_scenario=s, analysis_time=t0, t=first(representative_time_slice(m, t)))
+                ]
+                * demand[(node=ng, stochastic_scenario=s, analysis_time=t0, t=first(representative_time_slice(m, t)))]
+                for (n, s, t) in node_injection_indices(
+                    m; node=n, stochastic_scenario=s, t=t, temporal_block=anything
+                )
+                for ng in groups(n);
+                init=0,
+            )                                  
+           
+            # Commodity flows to storage units
+            - sum(
+                unit_flow[u, n, d, s, t_short]
+                for (u, n, d, s, t_short) in unit_flow_indices(
+                    m;
+                    node=n,
+                    direction=direction(:from_node),
+                    stochastic_scenario=s,
+                    t=t,
+                    temporal_block=anything,
```

Comment:
temporal_block=anything? Does this mean we don't want to use representative_periods for this expression?